tests/core/types/test_tx_wrappers.py

```diff
@@ -0,0 +1,222 @@
+
+        # Check return value
+        assert ret == "ret"
+
+        # Check internals
+        mock_declare.assert_called_once_with(
+            transaction=wrapper.tx,
+            signer=wrapper.account.signer,
+            alias=wrapper.alias,
+            overriding_path=wrapper.overriding_path,
+            watch_mode=watch_mode,
+        )
+
+
+@pytest.mark.asyncio
+@pytest.mark.parametrize("watch_mode", [None, "track", "debug"])
+@patch("nile.core.types.tx_wrappers.deploy_contract", return_value="ret")
+async def test_deploy_contract_wrapper_execute(
+    mock_deploy_contract,
+    watch_mode,
+):
+    account = await MockAccount(KEY, NETWORK)
+    with patch(
+        "nile.core.types.transactions.InvokeTransaction._get_tx_hash"
+    ) as mock_get_tx_hash:
+        mock_get_tx_hash.return_value = TX_HASH
+
+        tx = InvokeTransaction()
+        wrapper = DeployContractTxWrapper(
+            tx,
+            account,
+            alias="alias",
+            contract_name="contract",
+            overriding_path="path",
+            predicted_address=0x345,
+        )
+
+        assert wrapper.tx == tx
+        assert wrapper.account == account
+        assert wrapper.alias == "alias"
+        assert wrapper.overriding_path == "path"
+        assert wrapper.contract_name == "contract"
+        assert wrapper.predicted_address == 0x345
+
+        ret = await wrapper.execute(watch_mode=watch_mode)
+
+        # Check return value
+        assert ret == "ret"
+
+        # Check internals
+        mock_deploy_contract.assert_called_once_with(
+            transaction=wrapper.tx,
+            signer=wrapper.account.signer,
+            contract_name=wrapper.contract_name,
+            alias=wrapper.alias,
+            predicted_address=wrapper.predicted_address,
+            overriding_path=wrapper.overriding_path,
+            abi=None,
+            watch_mode=watch_mode,
+        )
+
+
+@pytest.mark.asyncio
+@pytest.mark.parametrize("watch_mode", [None, "track", "debug"])
+@patch("nile.core.types.transactions.get_contract_class", return_value="ContractClass")
+@patch("nile.core.types.transactions.get_class_hash", return_value=777)
+async def test_deploy_account_wrapper_execute(
+    mock_get_class_hash,
+    mock_get_contract_class,
+    watch_mode,
+):
+    account = await MockAccount(KEY, NETWORK)
+    with patch(
+        "nile.core.types.transactions.DeployAccountTransaction._get_tx_hash"
+    ) as mock_get_tx_hash:
+        mock_get_tx_hash.return_value = TX_HASH
+
+        tx = DeployAccountTransaction()
+        wrapper = DeployAccountTxWrapper(
+            tx,
+            account,
+            alias="alias",
+            overriding_path="path",
+        )
+
+        assert wrapper.tx == tx
+        assert wrapper.account == account
+        assert wrapper.alias == "alias"
+        assert wrapper.overriding_path == "path"
+
+        # Pending implementation
+        # ret = await wrapper.execute(watch_mode=watch_mode)
```

Comment:
What's blocking this?

Reply:
Whoops. I just forgot :). I will finish it.